checkov/common/proxy/proxy_client.py

```diff
@@ -31,3 +31,13 @@ def send_request(self, request: requests.Request) -> requests.Response:
 def call_http_request_with_proxy(request: requests.Request) -> Any:
     proxy_client = ProxyClient()
     return proxy_client.send_request(request=request)
+
+
+def get_proxy_envs() -> Mapping[str, str] | None:
```

Comment:
why `Mapping` instead of `dict`? TBH I'm not familiar with the exact difference, but didn't see it used anywhere else

Reply:
because its the type expected in the clon_from function in gitPython